apps/routing/script/controllers/productDetailsController.js

```diff
@@ -0,0 +1,30 @@
+app.controller('productDetailsController', 
+	
+	['$scope','$http', '$routeParams',function fileHistoryController($scope, $http, $routeParams) {
+    
+    	$scope.productId = $routeParams.productId;
+
+     	$http.get('data/products.json')
```

Comment:
Refactor options to avoid calling full product list with all details- 
1. Make http request using selected product id and get details of only that product.
2. Pass the selected product's details to this page (as previous page has that data)